beacon-chain/blockchain/new_slot.go

```diff
@@ -69,6 +68,12 @@ func (s *Service) NewSlot(ctx context.Context, slot types.Slot) error {
 				return err
 			}
 			s.store.SetJustifiedCheckptAndPayloadHash(bj, h)
+			if err := s.cfg.ForkChoiceStore.UpdateJustifiedCheckpoint(bj); err != nil {
+				return err
+			}
+			if err := s.cfg.ForkChoiceStore.UpdateFinalizedCheckpoint(f); err != nil {
+				return err
+			}
```

Comment:
I don't think we need to update finalized checkpt here. Are we doing it to be safe?

Reply:
I thought of the case when there's no block into the new epoch. But you may be right that it's overkill